storage/middleware.go

```diff
@@ -0,0 +1,51 @@
+package storage
+
+import (
+	"net/http"
+	"time"
+)
+
+const EntityCache = "MARKET_CACHE"
+
+type (
+	CacheResponse struct {
+		Status int
+		Header http.Header
+		Data   []byte
+	}
+
+	CacheData struct {
+		Data    CacheResponse
+		Expired int64
+	}
```

Comment:
we have multiple types of cache in this application and even this package. So `CacheData` is a bit overloaded here.

Reply:
Can you suggest something more suitable, please?